zone/groups.cpp

```diff
@@ -114,91 +114,85 @@ Group::~Group()
 
 //Split money used in OP_Split (/split and /autosplit).
 void Group::SplitMoney(uint32 copper, uint32 silver, uint32 gold, uint32 platinum, Client *splitter) {
-	//avoid unneeded work
-	if (
-		!copper &&
-		!silver &&
-		!gold &&
-		!platinum
-	) {
-		return;
-	}
-
-	uint8 member_count = 0;
-	for (uint32 i = 0; i < MAX_GROUP_MEMBERS; i++) {
-		// Don't split with Mercs or Bots
-		if (members[i] && members[i]->IsClient()) {
-			member_count++;
-		}
-	}
-
-	if (!member_count) {
-		return;
-	}
-
-	uint32 modifier;
-	if (member_count > 1) {
-		modifier = platinum % member_count;
-
-		if (modifier) {
-			platinum -= modifier;
-			gold += 10 * modifier;
-		}
-
-		modifier = gold % member_count;
-
-		if (modifier) {
-			gold -= modifier;
-			silver += 10 * modifier;
-		}
-
-		modifier = silver % member_count;
-
-		if (modifier) {
-			silver -= modifier;
-			copper += 10 * modifier;
-		}
-	}
-
-	auto copper_split = copper / member_count;
-	auto silver_split = silver / member_count;
-	auto gold_split = gold / member_count;
-	auto platinum_split = platinum / member_count;
-
-	for (uint32 i = 0; i < MAX_GROUP_MEMBERS; i++) {
-		if (members[i] && members[i]->IsClient()) { // If Group Member is Client
-			members[i]->CastToClient()->AddMoneyToPP(
-				copper_split,
-				silver_split,
-				gold_split,
-				platinum_split,
-				true
-			);
-
-			if (player_event_logs.IsEventEnabled(PlayerEvent::SPLIT_MONEY)) {
-				auto e = PlayerEvent::SplitMoneyEvent{
-					.copper = copper_split,
-					.silver = silver_split,
-					.gold = gold_split,
-					.platinum = platinum_split,
-					.player_money_balance = members[i]->CastToClient()->GetCarriedMoney(),
-				};
-
-				RecordPlayerEventLogWithClient(members[i]->CastToClient(), PlayerEvent::SPLIT_MONEY, e);
-			}
-
-			members[i]->CastToClient()->MessageString(
-				Chat::MoneySplit,
-				YOU_RECEIVE_AS_SPLIT,
-				Strings::Money(
-					platinum_split,
-					gold_split,
-					silver_split,
-					copper_split
-				).c_str()
-			);
-		}
-	}
+    // Return early if no money to split.
+    if (!copper && !silver && !gold && !platinum) {
+        return;
+    }
+
+    uint8 member_count = 0;
+    uint8 splitter_index = MAX_GROUP_MEMBERS
+    // Count the number of actual players in the group.
+    for (uint32 i = 0; i < MAX_GROUP_MEMBERS; i++) {
+        if (members[i] && members[i]->IsClient()) {
+            member_count++;
+            if (splitter && members[i]->CastToClient()->CharacterID() == splitter->CharacterID()) {
+                splitter_index = i;
+            }
+        }
+    }
```

Comment:
Can do this instead.
```cpp
for (const auto& m : members) {
	if (m && m->IsClient()) {
		member_count;
	}
}
```

Then just use the `splitter` Client that you're passing to the method instead of using `splitter_index`.